### PR TITLE
Feat: add config management apis

### PR DESCRIPTION
**Pull request by zzxwill**
Added some APIs for config management

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/apiserver/rest/webservice/config.go [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+		Param(ws.PathParameter("configType", "identifier of the config type").DataType("string")).
+		Returns(200, "OK", apis.ConfigType{}).
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Writes(apis.ConfigType{}))
+
+	ws.Route(ws.POST("/{configType}").To(s.createConfig).
+		Doc("create or update a config").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.PathParameter("configType", "identifier of the config type").DataType("string")).
+		Reads(apis.CreateApplicationRequest{}).
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/config.go:69]: Is still CreateApplicationRequest?

Comment (zzxwill) [reply pkg/apiserver/rest/webservice/config.go]: Fixed. Thanks.

File: pkg/apiserver/rest/apis/v1/types.go [hunk 2]
```diff
@@ -382,6 +401,14 @@ type CreateApplicationRequest struct {
 	Component   *CreateComponentRequest `json:"component"`
 }
 
+// CreateConfigRequest is the request body to creates a config
+type CreateConfigRequest struct {
+	Name          string `json:"name" validate:"checkname"`
+	Project       string `json:"project" validate:"checkname"`
```
Comment (barnettZQG) [inline pkg/apiserver/rest/apis/v1/types.go:407]: Please remove the validate, it required value is not empty.

Add the alias param.

File: pkg/apiserver/rest/webservice/config.go [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+		Reads(apis.CreateConfigRequest{}).
+		Returns(200, "OK", apis.EmptyResponse{}).
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Returns(404, "Not Found", bcode.Bcode{}).
+		Writes(apis.EmptyResponse{}))
+
+	ws.Route(ws.GET("/{configType}/configs").To(s.getConfigs).
+		Doc("get configs from a config type").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Filter(s.rbacUseCase.CheckPerm("configType", "list")).
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/config.go:83]:
```suggestion
		Filter(s.rbacUseCase.CheckPerm("config", "list")).
``

File: pkg/apiserver/rest/apis/v1/types.go (hunk 2 above)
Comment (zzxwill) [reply pkg/apiserver/rest/apis/v1/types.go]: Done.

**Review by barnettZQG — APPROVED** [review]
LGTM

File: vela-templates/definitions/internal/component/config-helm-repository.cue [hunk 4]
```diff
@@ -0,0 +1,41 @@
+"config-helm-repository": {
+	type: "component"
+	annotations: {
+		"alias.config.oam.dev": "Helm Repository"
+	}
+	labels: {
+		"catalog.config.oam.dev":       "velacore-config"
+		"type.config.oam.dev":          "helm-repository"
+		"multi-cluster.config.oam.dev": "true"
```
Comment (FogDong) [inline vela-templates/definitions/internal/component/config-helm-repository.cue:9]: Please add `ui-hidden： "true"`

Comment (zzxwill) [reply vela-templates/definitions/internal/component/config-helm-repository.cue:9]: Sure. I will also move it to addon fluxcd as what you did for dex addon.

Comment (barnettZQG) [reply vela-templates/definitions/internal/component/config-helm-repository.cue:9]: remove this definition from the core. I have already added it to the flux addon. https://github.com/oam-dev/catalog/pull/304

**Review by wonderflow — APPROVED** [review]

**Pull request merged**